pymc3/step_methods/hmc/quadpotential.py

```diff
@@ -88,13 +88,19 @@ def __init__(self, v):
         self.inv_s = 1. / s
         self.v = v
 
-    def velocity(self, x):
+    def velocity(self, x, out=None):
+        if out is not None:
+            out[:] = self.v
+            out[:] *= x
+            return
         return self.v * x
 
     def random(self):
         return floatX(normal(size=self.s.shape)) * self.inv_s
 
-    def energy(self, x):
+    def energy(self, x, velocity=None):
+        if velocity is not None:
+            return 0.5 * scipy.linalg.blas.ddot(x, velocity)
```

Comment:
this is `x.dot(velocity)`, but modestly extra faster, I guess? (just ran a silly benchmark, and it seems like it is ~30% faster)